stack/base/Dockerfile

```diff
@@ -16,20 +16,21 @@ WORKDIR /opt/
 
 ARG AIIDA_VERSION
 
+# Pin shared requirements in the base environemnt.
+# The lower bound of aiida-core is set to the version of aiida-core to prevent
+# lower aiida-core DB version installed and trigger database downgrade.
+RUN echo "pip==23.0.1" > /opt/requirements.txt && \
```

Comment:
Let's also try to bump pip to the latest version while we are here. (I am honestly not sure if we need this constrain anymore, but I guess it is safer to have, but we should try to update it regularly. New pip version in particular should have a much faster dependency resolver.

Otherwise I agree with the minimum aiida-core constrain. For the future we should still think whether to use an exact constraint. It can still happen that one app upgrades aiida-core, and another one downgrades it and breaks the DB.

Reply:
> bump pip to the latest version while we are here.

I remember there were some setuptools issues that blocked up updating it. I bump it and let's see if all works.

Reply:
Change it to `23.3.1` and all tests passed.